src/cdk/app.ts

```diff
@@ -32,27 +33,20 @@ export class LambdaService extends cdk.Stack {
             functionName,
         });
 
-        // If you need access to parameter store then uncomment and adjust the following:
-        // handler.addToRolePolicy(
-        //     new iam.PolicyStatement({
-        //         effect: iam.Effect.ALLOW,
-        //         resources: ['arn:aws:ssm:eu-west-1:[your account ID goes here]:parameter/*'],
-        //         actions: ['ssm:GetParameter'],
-        //     }),
-        // );
-
         const restApiName = `contributions-service-${stage.value}`;
 
-        // tslint:disable-next-line: no-unused-expression
-        new apigateway.LambdaRestApi(this, 'contributions-service', {
-            restApiName,
+        // https://aws.amazon.com/blogs/compute/announcing-http-apis-for-amazon-api-gateway/
+        const api = new apigatewayv2.CfnApi(this, 'contributions-service', {
+            name: restApiName,
             description: 'Service to serve contributions components',
-            proxy: true,
-            handler,
-            deployOptions: {
-                loggingLevel: apigateway.MethodLoggingLevel.INFO,
-                dataTraceEnabled: true,
-            },
+            protocolType: 'HTTP',
+            target: handler.functionArn,
+        });
+
+        handler.addPermission('grant-invoke', {
+            action: 'lambda:InvokeFunction',
+            principal: new iam.ServicePrincipal('apigateway.amazonaws.com'),
+            sourceArn: `arn:aws:execute-api:eu-west-1:${this.account}:${api.ref}*`,
```

Comment:
🎉 Great! So we're able to scope this with `api.ref`?

Reply:
yep! I was using `/$default/*` before which I thought should work but apparently not.